sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs

```diff
@@ -189,12 +189,12 @@ internal static TokenCredential CreateCredential(IConfiguration configuration)
                     string.IsNullOrWhiteSpace(clientId) ||
                     string.IsNullOrWhiteSpace(azureCloud))
                 {
-                    throw new ArgumentException("For managed identity as a federated identity credential, 'tenantId', 'clientId', 'azureCloud', and one of ['managedIdentityClientId', 'resourceId', 'objectId'] must be specified via environment variables or the configuration.");
+                    throw new ArgumentException("For managed identity as a federated identity credential, 'tenantId', 'clientId', 'azureCloud', and one of ['managedIdentityClientId', 'managedIdentityResourceId', 'managedIdentityObjectId'] must be specified via environment variables or the configuration.");
                 }
 
                 if (!string.IsNullOrWhiteSpace(resourceId))
                 {
-                    return new ManagedFederatedIdentityCredential(tenantId, clientId, new ResourceIdentifier(resourceId), azureCloud, additionallyAllowedTenantsList);
+                    return new ManagedFederatedIdentityCredential(tenantId, clientId, ManagedIdentityId.FromUserAssignedResourceId(new ResourceIdentifier(resourceId)), azureCloud, additionallyAllowedTenantsList);
```

Comment:
I'm assuming this and the change on L207 are just shortcuts that I didn't know about to normalize the parameter type for the constructor, yeah?